### PR TITLE
Clang-tidy - some automated fixes 

### PR DESCRIPTION
**Pull request by KipHamiltons**
I have started going through automated changes clang-tidy suggests. Here is a nice chunk of them.

Should wait for #697 to merge.

### CODE REVIEW

File: shared/utility/motion/splines/SmoothSpline.hpp [hunk 1]
```diff
@@ -90,7 +90,7 @@ namespace utility::motion::splines {
         /**
          * Inherit Load Points
          */
-        virtual void importCallBack() override {
+        void importCallBack() override {
```
Comment (CMurtagh-LGTM) [inline shared/utility/motion/splines/SmoothSpline.hpp:93]: I would wonder why virtual was here in the first place.

**Review by CMurtagh-LGTM — APPROVED** [review]
I have my trust in clang that it knows how to `std::move` properly.

File: module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp [hunk 2]
```diff
@@ -225,7 +225,7 @@ namespace module::behaviour::strategy {
                         // behaviour for overtime.
                         case GameMode::NORMAL:
                         case GameMode::OVERTIME: normal(gameState, phase, fieldDescription, field, ball); break;
-                        default: log<NUClear::WARN>("Game mode unknown.");
+                        default: log<NUClear::WARN>("Game mode BehaviourTeam.");
```
Comment (TrentHouliston) [inline module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp:228]: Wut?

File: module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp [hunk 3]
```diff
@@ -251,7 +251,7 @@ namespace module::behaviour::strategy {
             case Phase::PLAYING: penaltyShootoutPlaying(field, ball); break;  // Either kicking or goalie.
             case Phase::TIMEOUT: penaltyShootoutTimeout(); break;             // A pause in playing. Not in simulation.
             case Phase::FINISHED: penaltyShootoutFinished(); break;  // Happens when penalty shootout completely ends.
-            default: log<NUClear::WARN>("Unknown penalty shootout gamemode phase.");
+            default: log<NUClear::WARN>("BehaviourTeam penalty shootout gamemode phase.");
```
Comment (TrentHouliston) [inline module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp:254]: Also Wut?

File: module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp [hunk 4]
```diff
@@ -272,7 +272,7 @@ namespace module::behaviour::strategy {
             case Phase::PLAYING: normalPlaying(field, ball, fieldDescription); break;
             case Phase::FINISHED: normalFinished(); break;  // Game has finished.
             case Phase::TIMEOUT: normalTimeout(); break;    // A pause in playing. Not in simulation.
-            default: log<NUClear::WARN>("Unknown normal gamemode phase.");
+            default: log<NUClear::WARN>("BehaviourTeam normal gamemode phase.");
```
Comment (TrentHouliston) [inline module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp:275]: It's a BehaviourTeam phase!

**Review by TrentHouliston — CHANGES_REQUESTED** [review]
I noticed these, this is not a full review.

File: module/platform/darwin/HardwareIO/src/darwin/UART.cpp [hunk 5]
```diff
@@ -133,17 +133,17 @@ namespace Darwin {
...
-            if (bytesWritten < (int) count) {
+            if (bytesWritten < static_cast<int>(count)) {
                 reconnect();
                 reconnects++;
             }
         }
 
         if (reconnects > 0) {
-            std::cout << "Bytes Written: " << (int) bytesWritten << " Reconnects: " << (int) reconnects << "\r"
+            std::cout << "Bytes Written: " << bytesWritten << " Reconnects: " << static_cast<int>(reconnects) << "\r"
```
Comment (TrentHouliston) [inline module/platform/darwin/HardwareIO/src/darwin/UART.cpp:146]:
```suggestion
            std::cout << "Bytes Written: " << bytesWritten << " Reconnects: " << int(reconnects) << "\r"
```

File: module/platform/darwin/HardwareIO/src/darwin/UART.cpp [hunk 6]
```diff
@@ -157,10 +157,10 @@ namespace Darwin {
         uint8_t reconnects = 0;
         int bytesRead      = 0;
 
-        while ((bytesRead != (int) count) && (reconnects < 3)) {
+        while ((bytesRead != static_cast<int>(count)) && (reconnects < 3)) {
```
Comment (TrentHouliston) [inline module/platform/darwin/HardwareIO/src/darwin/UART.cpp:160]:
```suggestion
        while ((bytesRead != int(count)) && (reconnects < 3)) {
```

File: module/platform/darwin/HardwareIO/src/darwin/UART.cpp [hunk 7]
```diff
@@ -157,10 +157,10 @@ namespace Darwin {
...
             bytesRead = read(fd, buf, count);
 
-            if ((errno == EAGAIN) || ((bytesRead < (int) count) && (bytesRead > 0))) {
+            if ((errno == EAGAIN) || ((bytesRead < static_cast<int>(count)) && (bytesRead > 0))) {
```
Comment (TrentHouliston) [inline module/platform/darwin/HardwareIO/src/darwin/UART.cpp:163]:
```suggestion
            if ((errno == EAGAIN) || ((bytesRead < int(count)) && (bytesRead > 0))) {
```

File: module/platform/darwin/HardwareIO/src/darwin/UART.cpp [hunk 8]
```diff
@@ -171,9 +171,9 @@ namespace Darwin {
         }
 
         if (reconnects > 0) {
-            std::cout << "Bytes Read: " << (int) bytesRead << " Reconnects: " << (int) reconnects << " Data: ";
+            std::cout << "Bytes Read: " << bytesRead << " Reconnects: " << static_cast<int>(reconnects) << " Data: ";
```
Comment (TrentHouliston) [inline module/platform/darwin/HardwareIO/src/darwin/UART.cpp:174]:
```suggestion
            std::cout << "Bytes Read: " << bytesRead << " Reconnects: " << int(reconnects) << " Data: ";
```

File: module/platform/darwin/HardwareIO/src/darwin/UART.cpp [hunk 9]
```diff
@@ -171,9 +171,9 @@ namespace Darwin {
...
             for (int i = 0; i < bytesRead; i++) {
-                std::cout << (int) (*((uint8_t*) buf + i)) << " ";
+                std::cout << static_cast<int>(*(static_cast<uint8_t*>(buf) + i)) << " ";
```
Comment (TrentHouliston) [inline module/platform/darwin/HardwareIO/src/darwin/UART.cpp:176]:
```suggestion
                std::cout << int(*(static_cast<uint8_t*>(buf) + i)) << " ";
```

File: module/platform/darwin/HardwareIO/src/darwin/UART.cpp [hunk 10]
```diff
@@ -243,8 +243,8 @@ namespace Darwin {
         // length
         int length = 0;
         if (result.header.length < 2) {
-            std::cout << "Length: " << (int) result.header.length << ", " << (int) (result.header.length - 2) << "\r"
-                      << std::endl;
+            std::cout << "Length: " << static_cast<int>(result.header.length) << ", " << (result.header.length - 2)
```
Comment (TrentHouliston) [inline module/platform/darwin/HardwareIO/src/darwin/UART.cpp:246]:
```suggestion
            std::cout << "Length: " << int(result.header.length) << ", " << (result.header.length - 2)
```

File: shared/utility/motion/splines/SmoothSpline.hpp (hunk 1 above)
Comment (TrentHouliston) [reply shared/utility/motion/splines/SmoothSpline.hpp:93]: If it's using override, it's virtual.
The clang-tidy rule that is being applied is if it's override the virtual does nothing

**Review by TrentHouliston — CHANGES_REQUESTED** [review]

File: module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp (hunk 2 above)
Comment (KipHamiltons) [reply module/behaviour/strategy/SoccerStrategy/src/SoccerStrategy.cpp:228]: hahahahhahaha yikes. ctrl + d has done me dirty ahahahaha whoops

File: module/input/NatNet/src/NatNet.cpp [hunk 11]
```diff
@@ -61,32 +61,33 @@ namespace module::input {
...
+                    on<UDP::Multicast>(multicastAddress, dataPort)
+                        .then("NatNet Data", [this](const UDP::Packet& packet) {
+                            // Test if we are "connected" to this remote
+                            // And if we are we can use the data
+                            if (remote == packet.remote.address && version != 0) {
+                                process(packet.payload);
+                            }
+                            // We have started connecting but haven't received a return ping
+                            else if (remote == packet.remote.address && version == 0) {
+                                // TODO(unknown): maybe set a timeout here to try again
```
[the remark this answers is not in the document]
Comment (KipHamiltons) [reply module/input/NatNet/src/NatNet.cpp:77]: I had no idea who to assign this one to  tbh. Open to ideas - hardware perhaps?

File: module/input/NatNet/src/NatNet.cpp [hunk 12]
```diff
@@ -185,7 +186,7 @@ namespace module::input {
         mocap->recording              = (params & 0x01) == 0x01;
         mocap->tracked_models_changed = (params & 0x01) == 0x02;
 
-        // TODO there is an eod thing here
+        // TODO(unknown): there is an eod thing here
```
[the remark this answers is not in the document]
Comment (KipHamiltons) [reply module/input/NatNet/src/NatNet.cpp:189]: same with all the `NatNet` ones

File: module/platform/darwin/HardwareIO/src/darwin/UART.cpp (hunk 5 above)
Comment (KipHamiltons) [reply module/platform/darwin/HardwareIO/src/darwin/UART.cpp]: I added this suggestion, but I am curious - what's wrong with `static_cast`? I like it :(

Comment (TrentHouliston) [reply module/platform/darwin/HardwareIO/src/darwin/UART.cpp]: So many letters! It constantly makes things split one-liners into multi-liners!

**Review by CMurtagh-LGTM — DISMISSED** [review]

File: module/input/NatNet/src/NatNet.cpp (hunk 11 above)
[the remark this answers is not in the document]
Comment (ysims) [reply module/input/NatNet/src/NatNet.cpp:77]: Yeah probs hardware, it's the closest.

**Review by TrentHouliston — DISMISSED** [review]

File: module/localisation/BallLocalisation/src/BallLocalisation.cpp [hunk 13]
```diff
@@ -126,9 +125,9 @@ namespace module::localisation {
         on<Trigger<ResetBallHypotheses>, With<Sensors>, With<Field>, With<FieldDescription>, Sync<BallLocalisation>>()
             .then("Reset Ball Hypotheses",
                   [this](const ResetBallHypotheses& locReset,
-                         const Sensors& sensors,
+                         const Sensors& /*sensors*/,
```
Comment (ysims) [inline module/localisation/BallLocalisation/src/BallLocalisation.cpp:128]: Why is this here

Comment (KipHamiltons) [reply module/localisation/BallLocalisation/src/BallLocalisation.cpp:128]: unused parameter

Comment (ysims) [reply module/localisation/BallLocalisation/src/BallLocalisation.cpp:128]: Why have it there at all?

Comment (KipHamiltons) [reply module/localisation/BallLocalisation/src/BallLocalisation.cpp:128]: Well shit. Very nice catch!!

**Review by ysims — APPROVED** [review]
yeah looks good
